aten/src/ATen/native/sparse/cuda/SparseCUDATensor.cpp

```diff
@@ -46,7 +46,11 @@ SparseTensor& sparse_mask_out_cuda(SparseTensor& r, const Tensor& t, const Spars
     view_size[d + 1] = mask.size(mask.sparse_dim() + d);
   }
 
-  Tensor t_view = t.view(view_size);
+  Tensor t_view;
```

Comment:
Isn't this if-else equivalent to a single reshape call?

Reply:
Functionally, yes, but this is slightly faster.

Reply:
Reshape does more checks?

Reply:
Reshape goes through 2 rounds of dispatch (for reshape and for view that reshape calls internally), and checks done by reshape are then repeated in view. Unless you care about sum-us overheads in your user code, you can safely ignore this and do reshape everywhere, but in the core I wanted the most performant approach.

Reply:
Thanks for the insight :)

Reply:
@ngimel Can we file an issue about this? I feel that `contiguous()` should not go through a dispatch at all when the input is contiguous, so `t.contiguous().view(view_size)` should be an equivalent way to write this code. And then if that's the case, `reshape()` should also skip dispatch if the input is contiguous. This is predicated on being able to define some function completely inline bypassing dispatch though.

Reply:
Filed #31303 for `.contiguous()`. 
It is more convoluted with `reshape()`. I could (and I did) make `.view()` and `.reshape()` go through the same no-dispatch helper function. However, derivative for `.reshape()` is deliberately not defined https://github.com/pytorch/pytorch/blob/master/tools/autograd/derivatives.yaml#L725-L730, so `.reshape()` can be only a compound function and can call only functions with defined derivatives inside, thus my approach failed, and I don't see any alternative to `.reshape()` calling `.view()` internally.

Reply:
Err, I can believe that reshape is more complicated to deal with, but what's wrong with it calling view() internally? That's what I'd expect.